### ISSUE TITLE
NaN is detected on backward computation of ConnectionistTemporalClassification

### ISSUE THREAD

**Issue by synetkim** [chainer/chainer#3529]
Hello.

After several epochs, NaN occurred on backward computation of CTC. 
My training loop and optimizer is like below:

opt = optimizers.AdaDelta(rho=0.95, eps=1e-8)

    -. training Epoch loop
        ...
        opt.setup(model)
        opt.add_hook(chainer.optimizer.GradientClipping(2))
        ...
       -.  training bach loop
             ...
            <forward process> 
             ...
            opt.target.cleargrads()
            loss.backward()
            loss.unchain_backward()
            opt.update()
            loss = chainer.Variable(xp.zeros((), dtype=np.float32))

* Conditions
  - Chainer version: 3.0.0rc1
  - CuPy version: 1.0.3
  - OS/Platform: Ubuntu14.04
  - CUDA/cuDNN version: 8.0 / 6.0
  - etc.

Thank you for your cooperation!

**Comment by HawkAaron** [comment]
Hello,

Could you tell we how to truncate the loss bp in LSTM?

**Comment by Crissman** [comment]
Thanks for your issue report. NaN can happen with CTC due to the data. http://musyoku.github.io/2017/06/16/Connectionist-Temporal-Classification/

Can you provide data and code to replicate the issue?

**Comment by mitmul** [comment]
Hi @synetkim, it depends on the data you used, so could you provide data and code to replicate the issue?

**Comment by kmaehashi** [comment]
Feel free to reopen the issue if you still have a problem.
